src/components/common/fields/Autocomplete.vue

```diff
@@ -0,0 +1,163 @@
+<template>
+  <div 
+    class="autocomplete"
+    v-if="dropdownShown"
+  >
+    <button
+      class="autocomplete-option"
+      :class="{ 'autocomplete-option--active': item[autocompleteType] === hoveredValue }"
+      v-for="item in list"
+      :key="item[autocompleteType]"
+      :id="item[autocompleteType]"
+      :ref="item[autocompleteType]"
+      @mousedown="emitDataToSetInputValue"
+      @mouseover="hoveredValue = item[autocompleteType]"
+    >
+      {{ item[autocompleteType] }}
+    </button>
+  </div>
+</template>
+
+<script>
+import { clearObject } from '@/utils/clearObject'
+import { ApiCallerFactory } from '@/api-caller-factory'
+import { ErrorHandler } from '@/utils/ErrorHandler'
+import _ from 'lodash'
+
+export default {
+  props: {
+    inputValue: { type: String, default: '' },
+    autocompleteType: { type: String, default: '' },
+    shouldShowDropdown: { type: Boolean, default: false }
+  },
+
+  components: {
+    // components
+  },
+
+  data () {
+    return {
+      hoveredValue: '',
+      list: [],
+      closeDropdown: false
+    }
+  },
+
+  computed: {
+    dropdownShown () {
+      if (this.shouldShowDropdown) {
+        this.getList()
+        window.addEventListener('keydown', this.onKey)
+        window.addEventListener('mouseup', this.onClick)
+        this.closeDropdown = false
+      }
+      if (this.closeDropdown) {
+        window.removeEventListener('mouseup', this.onClick, false)
+        window.removeEventListener('keydown', this.onKey, false)
+      }
+      return this.shouldShowDropdown | !this.closeDropdown
+    }
+  },
+
+  created () {
+    // created
+  },
+
+  methods: {
+
+    async getList () {
+      let response = []
+      if (this.inputValue || this.enteredAddress) {
+        try {
+          response = await ApiCallerFactory
+            .createCallerInstance()
+            .getWithSignature('/identities', {
+              filter: clearObject({
+                [this.autocompleteType]: this.inputValue
+              })
+            })
+          this.list = response.data
+        } catch (error) {
+          ErrorHandler.processWithoutFeedback(error)
+        }
+      } else {
+        this.list = response
+      }
+    },
+
+    setHoverOnKeyPress (element) {
+      this.hoveredValue = element.id
+      element.focus()
+      event.preventDefault()
+    },
+
+    onKey (event) {
+      const currentElement = this.hoveredValue ? document.getElementById(this.hoveredValue) : undefined
+      const hasNextSibling = currentElement && currentElement.nextSibling
```

Comment:
@max-nes If we have no currentElement and ArrowUp or ArrowDown pressed, then we need to set first dropdown option as currentElement. I renamed currentElement to currentHoveredOption, to avoid confusion